pkg/handlers/generic/lifecycle/registrymirror/distribution/handler.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package distribution
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"text/template"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/pflag"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/lifecycle/addons"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/lifecycle/config"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/lifecycle/registrymirror/utils"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/handlers/options"
+	handlersutils "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/handlers/utils"
+)
+
+const (
+	DefaultHelmReleaseName      = "registry-mirror"
+	DefaultHelmReleaseNamespace = "registry-mirror-system"
+)
+
+type Config struct {
+	*options.GlobalOptions
+
+	defaultValuesTemplateConfigMapName string
+}
+
+func (c *Config) AddFlags(prefix string, flags *pflag.FlagSet) {
+	flags.StringVar(
+		&c.defaultValuesTemplateConfigMapName,
+		prefix+".default-values-template-configmap-name",
+		"default-distribution-registry-mirror-helm-values-template",
+		"default values ConfigMap name",
+	)
+}
+
+type Distribution struct {
+	client              ctrlclient.Client
+	config              *Config
+	helmChartInfoGetter *config.HelmChartGetter
+}
+
+func New(
+	c ctrlclient.Client,
+	cfg *Config,
+	helmChartInfoGetter *config.HelmChartGetter,
+) *Distribution {
+	return &Distribution{
+		client:              c,
+		config:              cfg,
+		helmChartInfoGetter: helmChartInfoGetter,
+	}
+}
+
+func (n *Distribution) Apply(
+	ctx context.Context,
+	_ v1alpha1.RegistryMirror,
+	cluster *clusterv1.Cluster,
+	log logr.Logger,
+) error {
+	log.Info("Applying Distribution registry mirror installation")
+
+	remoteClient, err := remote.NewClusterClient(
+		ctx,
+		"",
+		n.client,
+		ctrlclient.ObjectKeyFromObject(cluster),
+	)
+	if err != nil {
+		return fmt.Errorf("error creating remote cluster client: %w", err)
+	}
+
+	err = handlersutils.EnsureNamespaceWithMetadata(
```

Comment:
Why do we need to explicitly create helm release namespace? Does not HRP creates it? also are we expecting additional metadata on the namespace later? currently it is set to `nil`

Reply:
Ahh good eye, I copied this from a different handler that needs to install additional resources into this namespace.
Let me remove it here and then we can add it back in a follow up if we do need to guarantee that it exists.